src/VisualStudio/Core/Def/Watson/FaultReporter.cs

```diff
@@ -160,6 +175,39 @@ public static void ReportFault(Exception exception, FaultSeverity severity, bool
             }
         }
 
+        private static void UpdateBlamedMethod(IFaultUtility faultUtility, Exception exception)
+        {
+            var blamedMethod = faultUtility.GetBucketParameter(P5MethodNameDefaultIndex);
+
+            // We'll only override anything if the default logic blamed something we didn't want
+            if (!UnblameableMethodPrefixes.Any(p => blamedMethod.StartsWith(p)))
+            {
+                return;
+            }
+
+            // If anything fails here, we'll just keep the failure as is rather than potentially losing it
+            try
+            {
+                var stackTrace = new StackTrace(exception);
+                foreach (var stackFrame in stackTrace.GetFrames())
+                {
+                    var method = stackFrame.GetMethod();
+                    if (method != null && method.DeclaringType != null)
+                    {
+                        // Get the full name of the method, without parameters
+                        var methodName = method.DeclaringType.FullName + "." + method.Name;
+                        if (!UnblameableMethodPrefixes.Any(p => methodName.StartsWith(p)))
+                        {
+                            faultUtility.SetBucketParameter(P4ModuleNameDefaultIndex, method.DeclaringType.Assembly.GetName().Name);
```

Comment:
is the full stack trace still available in queries (in case for some reason the code in `Roslyn.Utilities.Contracts` is relevant)?

Reply:
Yep, the full stack trace is always available; this is just what gets picked for purposes of Watson cab submission and matching on dump rules.